src/components/dialogs/GifSelect.tsx

```diff
@@ -0,0 +1,404 @@
+    case 'show':
+      content = <GifList control={control} onSelectGif={onSelectGif} />
+      snapPoints = ['100%']
+      break
+    case 'hide':
+      content = <GiphyConsentNotGiven control={control} />
+      break
+    default:
+      content = <GiphyConsentPrompt control={control} />
+      break
+  }
+
+  return (
+    <Dialog.Outer
+      control={control}
+      nativeOptions={{sheet: {snapPoints}}}
+      onClose={onClose}>
+      <Dialog.Handle />
+      {content}
+    </Dialog.Outer>
+  )
+}
+
+function GifList({
+  control,
+  onSelectGif,
+}: {
+  control: Dialog.DialogControlProps
+  onSelectGif: (gif: Gif) => void
+}) {
+  const {_} = useLingui()
+  const t = useTheme()
+  const {gtMobile} = useBreakpoints()
+  const [search, setSearch] = useState('')
+
+  const isSearching = search.length > 0
+
+  const trendingQuery = useGiphyTrending()
+  const searchQuery = useGifphySearch(search)
+
+  const {
+    data,
+    fetchNextPage,
+    isFetchingNextPage,
+    hasNextPage,
+    error,
+    isLoading,
+    isError,
+    refetch,
+  } = isSearching ? searchQuery : trendingQuery
+
+  const flattenedData = useMemo(() => {
+    const uniquenessSet = new Set<string>()
+
+    function filter(gif: Gif) {
+      if (!gif) return false
+      if (uniquenessSet.has(gif.id)) {
+        return false
+      }
+      uniquenessSet.add(gif.id)
+      return true
+    }
+    return data?.pages.flatMap(page => page.data.filter(filter)) || []
+  }, [data])
+
+  const renderItem = useCallback(
+    ({item}: {item: Gif}) => {
+      return <GifPreview gif={item} onSelectGif={onSelectGif} />
+    },
+    [onSelectGif],
+  )
+
+  const onEndReached = React.useCallback(() => {
+    if (isFetchingNextPage || !hasNextPage || error) return
+    fetchNextPage()
+  }, [isFetchingNextPage, hasNextPage, error, fetchNextPage])
+
+  const hasData = flattenedData.length > 0
+
+  const onGoBack = useCallback(
+    () => (isSearching ? setSearch('') : control.close()),
+    [control, isSearching],
+  )
+
+  const listHeader = useMemo(() => {
+    return (
+      <>
+        <View
+          style={[
+            a.relative,
+            a.mb_lg,
+            a.flex_row,
+            a.align_center,
+            !gtMobile && isWeb && a.gap_md,
+          ]}>
+          {/* cover top corners */}
+          <View
+            style={[
+              a.absolute,
+              {top: 0, left: 0, right: 0, height: '50%'},
+              t.atoms.bg,
+            ]}
+          />
+
+          {!gtMobile && isWeb && (
+            <Button
+              size="small"
+              variant="ghost"
+              color="secondary"
+              shape="round"
+              onPress={() => control.close()}
+              label={_(msg`Close GIF dialog`)}>
+              <ButtonIcon icon={Arrow} size="md" />
+            </Button>
+          )}
+
+          <TextField.Root>
+            <TextField.Icon icon={Search} />
+            <TextField.Input
```

Comment:
What's up with our text input that allows this? Is this specific to our component or does the raw RN one have the same issue?

https://github.com/bluesky-social/social-app/assets/810438/905c1b0d-72c0-4551-b59a-7e65099bbf42

Reply:
Removing the `value` prop fixes this - a little concerning. We should check the app for other places where this might happen